src/main/scala/no/ndla/imageapi/swagger/ScalatraServletReaderExtension.scala

```diff
@@ -0,0 +1,49 @@
+/*
+ * Part of NDLA image_api.
+ * Copyright (C) 2017 NDLA
+ *
+ * See LICENSE
+ */
+
+package no.ndla.imageapi.swagger
+
+import java.lang.annotation.Annotation
+import java.lang.reflect.{Method, Type}
+
+import com.typesafe.scalalogging.LazyLogging
+import io.swagger.annotations.{Api, ApiOperation}
+import io.swagger.models.Operation
+import io.swagger.servlet.ReaderContext
+import io.swagger.servlet.extensions.{ReaderExtension, ServletReaderExtension}
+import io.swagger.util.{PathUtils, ReflectionUtils}
+
+
+class ScalatraServletReaderExtension extends ReaderExtension with LazyLogging {
```

Comment:
Den er delaktig i hvordan swagger-dokumentasjonen rendres. De bibliotekene som vi har dratt inn nå er laget for jax-rs og java. Defaulten i swagger-genereringen er at den bruker navnet på metoden som endepunkt. I tilfellene her, så ville da typisk endepunktet for å hente et bilde blitt `/image-api/v1/images/getImage` mens vi vil at det skal være `/image-api/v1/images/{image_id}`. Ved å bruke vår egen klasse til rendringen, så har vi overridet akkurat den funksjonaliteten og benytter `nickname` til å rendre path.

Denne klassen refereres fra `src/main/resources/META-INF/services/io.swagger.servlet.extension.ReaderExtension`. Vår klasse lastes da som en service-extension og benyttes til rendring.